tests/Findologic/Request/FilterHandlerTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\FinSearch\Tests\Findologic\Request;
+
+use FINDOLOGIC\Api\Requests\SearchNavigation\SearchRequest;
+use FINDOLOGIC\FinSearch\Findologic\Request\Handler\FilterHandler;
+use FINDOLOGIC\FinSearch\Findologic\Response\Xml21\Filter\ColorPickerFilter;
+use FINDOLOGIC\FinSearch\Findologic\Response\Xml21\Filter\Media;
+use FINDOLOGIC\FinSearch\Findologic\Response\Xml21\Filter\Values\ColorFilterValue;
+use FINDOLOGIC\FinSearch\Findologic\Response\Xml21\Filter\Values\FilterValue;
+use FINDOLOGIC\FinSearch\Struct\FiltersExtension;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Content\Product\Events\ProductSearchCriteriaEvent;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Event\ShopwareEvent;
+use Symfony\Component\HttpFoundation\Request;
+
+class FilterHandlerTest extends TestCase
+{
+    public function filterRequestProvider(): array
+    {
+        return [
+            'Filter name with special characters' => [
+                'filterName' => 'filter-name',
+                'filterValue' => sprintf('filter-name%ssomevalue', FilterValue::DELIMITER),
+                'expectedValue' => 'somevalue'
+            ],
+            'Filter name with spaces' => [
+                'filterName' => 'filter name',
+                'filterValue' => sprintf('filter name%ssomevalue', FilterValue::DELIMITER),
+                'expectedValue' => 'somevalue'
+            ],
+            'Filter value with special characters' => [
+                'filterName' => 'filtername',
+                'filterValue' => sprintf('filtername%ssome-value', FilterValue::DELIMITER),
+                'expectedValue' => 'some-value'
+            ],
+            'Filter value with space' => [
+                'filterName' => 'filtername',
+                'filterValue' => sprintf('filtername%ssome value', FilterValue::DELIMITER),
+                'expectedValue' => 'some value'
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider filterRequestProvider
+     */
+    public function testFilters(string $filterName, string $filterValue, string $expectedValue)
+    {
+        $searchNavigationRequest = new SearchRequest();
+        $expectedColorFilter = new ColorPickerFilter($filterName, $filterName);
+        $expectedColorFilter->addValue(
+            (new ColorFilterValue($filterValue, $filterValue, $filterName))
```

Comment:
These brackets look weirdly indented and at wrong places. Somehow at `setMedia` there are two closing brackets.